apps/src/code-studio/initApp/loadApp.js

```diff
@@ -282,23 +281,35 @@ async function loadAppAsync(appOptions) {
     return appOptions;
   }
 
-  if (appOptions.channel || isViewingStudentAnswer) {
+  // Special case -- If the level is not cached, not channel-backed, and the
+  // user is signed-out, load source code from session storage.
+  // TODO: Refactor the logic in this function so this isn't a special case
```

Comment:
@jamescodeorg is there a ticket for this?

Reply:
Thanks for the reminder, there is now!  https://codedotorg.atlassian.net/browse/LP-2141